exchanges/binance/binance_websocket.go

```diff
@@ -51,7 +51,7 @@ func (b *Binance) WsConnect() error {
 	dialer.Proxy = http.ProxyFromEnvironment
 	var err error
 	if b.Websocket.CanUseAuthenticatedEndpoints() {
-		listenKey, err = b.GetWsAuthStreamKey(context.TODO())
+		listenKey, err = b.GetWsAuthStreamKey(ctx)
```

Comment:
Just a quick comment, but you're definitely going to want to be passing context to `KeepAuthKeyAlive` for here and US to avoid disconnects. Hopefully this is the only additional thing you need to consider

Reply:
🦅 Good spotting, went and implemented most of what I could find.